src/Toolchain/ToolchainProvider.ts

```diff
@@ -113,62 +120,65 @@ export class ToolchainProvider implements vscode.TreeDataProvider<BaseNode> {
     }
   }
 
-  error(msg: string): void {
+  error(msg: string, ...args: string[]): Thenable<string|undefined> {
     Logger.error(this.tag, msg);
-    vscode.window.showErrorMessage(msg);
+    return vscode.window.showErrorMessage(msg, ...args);
   }
 
   refresh() {
     this._onDidChangeTreeData.fire();
   }
 
   install() {
-    showInstallQuickInput().then(
-        ([toolchainEnv, toolchain]) => {
-          // NOTE(jyoung)
-          // The `DebianToolchain` of the backend and the `DebianToolchain` of this project
-          // are not recognized as the same object by `instanceof` function.
-          const installed = toolchainEnv.listInstalled().filter(
-              value => value.constructor.name === 'DebianToolchain');
-
-          if (installed.length > 1) {
-            this.error('Installed debian toolchain must be unique');
-            return;
-          }
-
-          if (installed.length !== 1) {
-            toolchainEnv.install(toolchain).then(() => this.refresh(), () => {
-              this.error('Installation is failed');
-            });
-            return;
-          }
-
-          vscode.window
-              .showInformationMessage(
-                  'Do you want to remove the existing and re-install? Backend toolchain can be installed only once.',
-                  'Yes', 'No')
-              .then((answer) => {
-                if (answer === 'Yes') {
-                  const jobs: Array<Job> = [];
-                  jobs.push(new JobUninstall(installed[0].uninstall()));
-                  jobs.push(new JobInstall(toolchain.install()));
-                  const name = `${toolchain.info.name}-${toolchain.info.version ?.str()}`;
-                  toolchainEnv.request(jobs).then(
-                      () => {
-                        this.refresh();
-                        vscode.window.showInformationMessage(`Install ${name} successfully`);
-                      },
-                      () => {
-                        this.error('Installation is failed');
-                      });
-                } else {
-                  Logger.info(this.tag, 'Installation is canceled');
-                }
-              });
-        },
-        () => {
-          Logger.info(this.tag, 'Installation is canceled.');
-        });
+    const notifyInstalled = (toolchainEnv: ToolchainEnv, toolchain: Toolchain) => {
+      const name = `${toolchain.info.name}-${toolchain.info.version ?.str()}`;
+      vscode.window.showInformationMessage(`Install ${name} successfully`);
+      DefaultToolchain.getInstance().ask(toolchainEnv, toolchain).then(() => this.refresh());
```

Comment:
(optional)
It looks handy that user is asked whether to set the toolchain as `default toolchain` when installation is done.

I think users may expect the toolchain to remember the default toolchain between vscode's on and off.
So I suggest the furthur feature: remembering the default toolchain

Reply:
> So I suggest the furthur feature: remembering the default toolchain

I think so. I have a plan to save the default toolchain information in vscode. So I asked you about the related content a few days ago. :)
I made an issue. #1015

Reply:
I remember that. Now I understand the context. :-D